lib/lunchbot_web/live/component/shared_cart/shared_cart.ex

```diff
@@ -0,0 +1,75 @@
+defmodule LunchbotWeb.SharedCart do
+  @moduledoc false
+
+  use LunchbotWeb, :live_view
+
+  import Ecto.Query
+  import Ecto.Changeset
+  alias Lunchbot.LunchbotData.Office
+  alias Lunchbot.LunchbotData.OfficeLunchOrder
+  alias LunchbotWeb.SharedCart.OfficeSelector
+  alias LunchbotWeb.SharedCart.OfficeSelectorInterfaceContext
+
+  def mount(_params, _session, socket) do
+    if connected?(socket), do: Lunchbot.LunchbotData.OrderItem.subscribe()
+    {:ok, fetch(socket) |> assign_office() |> assign_changeset()}
+  end
+
+  def assign_office(socket) do
+    socket
+    |> assign(:office_selector, %OfficeSelector{})
+  end
+
+  def assign_changeset(%{assigns: %{office_selector: office_selector}} = socket) do
+    socket
+    |> assign(:changeset, OfficeSelectorInterfaceContext.change_office(office_selector))
+  end
+
+  # Fetch DB data, assign to socket watched by this LiveView
+  defp fetch(socket) do
+    office_lunch_orders = get_lunch_order_for_office()
+
+    offices =
+      Lunchbot.Repo.all(Office)
+      |> IO.inspect(label: :"all offices?")
+
+    assign(socket, office_lunch_orders: office_lunch_orders, offices: offices)
+  end
+
+  # listens to the notifications broadcast by OrderItems
+  # fetches a fresh listing of office lunch orders from the database
+  def handle_info({Lunchbot.LunchbotData.OrderItem, [:order_item, _], _}, socket) do
+    {:noreply, fetch(socket)}
+  end
+
+  # handles office changes
+  def handle_event("change-office", %{"office_selector" => formtest_params}, socket) do
+    IO.inspect(socket, label: "IO:socket")
+    IO.inspect(formtest_params, label: "IO:formtest_params")
+
+    {:noreply,
+     assign(socket,
+       office_lunch_orders: get_lunch_order_for_office(formtest_params["office_selection"])
+     )}
+  end
+
+  def get_lunch_order_for_office(office_id \\ 2, date \\ Date.utc_today()) do
+    filter_lunch_orders_to_today_and_onwards = from(o in OfficeLunchOrder, where: o.day >= ^date)
```

Comment:
I'm not sure why, but I've had this problem for a while now where trying to alias one of the structs like `OfficeLunchOrder` has not been working like this, I would always get a "struct undefined" error. If this is working on your end then that's good, just checking

Reply:
I've got the alias up top so its working fine. From my experience you will get the error you mentioned if you either don't have `Ecto.Query` imported or if both `Ecto.Query` and I think `Lunchbot.Repo` are imported (they both define `from()`)